axiom/tests/test_log_events.c

```diff
@@ -19,120 +20,138 @@
 #define LOG_MESSAGE_1 LOG_MESSAGE " 1"
 #define LOG_MESSAGE_2 LOG_MESSAGE " 2"
 #define LOG_MESSAGE_3 LOG_MESSAGE " 3"
+#define LOG_MESSAGE_4 LOG_MESSAGE " 4"
 #define LOG_TIMESTAMP 12345
 
 static nr_log_event_t* create_sample_event(const char* message) {
   nr_log_event_t* e = nr_log_event_create();
   nr_log_event_set_log_level(e, LOG_LEVEL);
   nr_log_event_set_message(e, message);
   nr_log_event_set_timestamp(e, LOG_TIMESTAMP * NR_TIME_DIVISOR_MS);
+  nr_log_event_set_priority(e, 0);
   return e;
 }
 
 static void test_events_success(void) {
   /* Normal operation */
-  nr_analytics_events_t* events = NULL;
+  nr_log_events_t* events = NULL;
   nr_log_event_t* e = NULL;
-  nr_random_t* rnd = nr_random_create_from_seed(12345);
-  const char* json;
+  void* test_e;
+  nr_vector_t* vector = NULL;
+  bool pass;
+  char* json;
   const char* expected;
 
   events = nr_log_events_create(10);
   tlib_fail_if_null("events created", events);
 
   e = create_sample_event(LOG_MESSAGE_0);
-  nr_log_events_add_event(events, e, rnd);
-  nr_log_event_destroy(&e);
+  nr_log_event_set_priority(e, 3);
+  nr_log_events_add_event(events, e);
 
   e = create_sample_event(LOG_MESSAGE_1);
-  nr_log_events_add_event(events, e, rnd);
-  nr_log_event_destroy(&e);
+  nr_log_event_set_priority(e, 2);
+  nr_log_events_add_event(events, e);
 
   e = create_sample_event(LOG_MESSAGE_2);
-  nr_log_events_add_event(events, e, rnd);
-  nr_log_event_destroy(&e);
+  nr_log_event_set_priority(e, 1);
+  nr_log_events_add_event(events, e);
 
   e = create_sample_event(LOG_MESSAGE_3);
-  nr_log_events_add_event(events, e, rnd);
-  nr_log_event_destroy(&e);
+  nr_log_event_set_priority(e, 0);
+  nr_log_events_add_event(events, e);
+
+  // minmax heap will store events like this (showing priorities)
+  //         0
+  //       3   2
+  //      1
+  //
+  // so when walking the heap and converting to a vector we expect
+  // the following order:
+  // LOG_MESSAGE_3 (pri 0)
+  // LOG_MESSAGE_0 (pri 3)
+  // LOG_MESSAGE_1 (pri 2)
+  // LOG_MESSAGE_2 (pri 1)
```

Comment:
```suggestion
  // LOG_MESSAGE_2 (pri 1)
  // clang-format on
```
Restore clang-format operation